src/services/drpcEvents/index.ts

```diff
@@ -241,4 +249,85 @@ export default class DrpcEvents {
       })
     }
   }
+
+  private async handleParentQuery(id: UUID, childQuery: QueryRow, params: SubmitQueryResponseRPCParams) {
+    const [parentQuery]: QueryRow[] = await this.db.get('query', { id })
+    if (!parentQuery) {
+      this.logger.warn('parent query not found %s', id)
+      return
+    }
+
+    // adding childs and parent emissions
+    const total: number = parseInt(childQuery.details.emissions) + parseInt(params.emissions || '0')
+    await this.db.update(
+      'query',
+      { id },
+      {
+        status: 'resolved',
+        details: {
+          ...parentQuery.details,
+          emissions: total.toString(),
+        },
+      }
+    )
+
+    const [connection] = await this.db.get('connection', { id: parentQuery.connection_id })
+    if (!connection.agent_connection_id) return this.handleError(childQuery, connection)
+
+    let rpcResponse: DrpcResponse
+    try {
+      const maybeResponse = await this.cloudagent.submitDrpcRequest(
```

Comment:
I think we're missing a query here to make sure that all queries that this is the parent of are in state `resolved`

Reply:
to be fair this might just cover the multi forwards excluding expiry date/timeout